File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from app.models.enums import NotificationType
from app.schemas.base import CamelModel


class NotificationOut(CamelModel):
    id: uuid.UUID
    type: NotificationType
    title: str
    message: str
    is_read: bool
    created_at: datetime


class NotificationListResponse(CamelModel):
    items: list[NotificationOut]
    unread_count: int


class UnreadCountResponse(CamelModel):
    unread_count: int

File: frontend/src/types/notification.ts
export type NotificationType =
  | "LEAVE_SUBMITTED"
  | "LEAVE_APPROVED"
  | "LEAVE_REJECTED"
  | "PAYROLL_UPDATED"
  | "ATTENDANCE_ALERT"
  | "PROFILE_UPDATED"
  | "SYSTEM";

export interface Notification {
  id: string;
  type: NotificationType;
  title: string;
  message: string;
  isRead: boolean;
  createdAt: string;
}

export interface NotificationList {
  items: Notification[];
  unreadCount: number;
}

export interface UnreadCount {
  unreadCount: number;
}
